downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,460 @@
+
+```
+int result = -9223372036854775808; // error: '9223372036854775808' is out of range for 'int'
+```
+
+#### Restriction on the typed binding pattern of an outer join clause to use `var`
+
+Previously, the typed binding pattern in an outer join clause was allowed to be any type descriptor, but now it has been restricted to allow only `var` so that an optional type (`T?`) will be inferred as the type.
+
+```ballerina
+type User record {|
+    readonly int id;
+    string name;
+|};
+
+type Login record {|
+    int userId;
+    string time;
+|};
+
+type Entry record {|
+    string? name;
+    string login;
+|};
+
+public function main() {
+    Login[] logins = [
+        {userId: 6789, time: "20:10:23"},
+        {userId: 1234, time: "10:30:02"},
+        {userId: 3987, time: "12:05:00"}
+    ];
+
+    table<User> key(id) users = table [
+            {id: 1234, name: "Keith"},
+            {id: 6789, name: "Anne"}
+        ];
+
+    Entry[] _ = from var login in logins
+        outer join var user in users
+        on login.userId equals user?.id
+        select {name: user?.name, login: login.time};
+
+    // Now, the code below gives an error at the typed binding pattern of the outer join clause.
+    _ = from var login in logins
+        outer join User user in users  // error: outer join must be declared with 'var'
+        on login.userId equals user.id
+        select {name: user?.name, login: login.time};
+}
+```
+
+## Compiler API updates
+
+### New features
+
+#### Semantic API
+- Added a new `annotAttachments()` API to get the annotation attachments and their constant values from the annotatable symbols
+- Introduced a new `ClientDeclSymbol` symbol to represent the semantic information of the client-declaration statement
+
+### Improvements
+
+#### Semantic API
+Improved the `constValue()` method to retrieve the constant value as an object from the constant symbol
+
+#### Syntax API
+
+Added a few methods to the `NodeParser` API. The methods below have been introduced for the `NodeParser` class.
+  - `Node parseObjectMember(String text)`
+  - `ModulePartNode parseModulePart(String text)`
+  - `IntermediateClauseNode parseIntermediateClause(String text, boolean allowActions)`
+
+### Backward-incompatible changes
+
+Updated the `NodeFactory` methods to allow optional terminating semicolons for module-level declarations. The methods below in the `NodeFactory` have been updated with an extra parameter for the optional semicolon token.
+
+  - `createServiceDeclarationNode`
+  - `createFunctionBodyBlockNode`
+  - `createEnumDeclarationNode`
+  - `createClassDefinitionNode`
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.3.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+label%3ATeam%2FCompilerFE+is%3Aclosed).
+
+## Compiler API updates
```

Comment:
Why are we having repetitive section named `Compiler API updates`?

Reply:
Is it incorrect to have `Language updates` and `Compiler updates` as 2 separate sections? @hasithaa @gimantha @MaryamZi

Reply:
Removed the duplicate section.